portability/memory.cc

```diff
@@ -339,7 +339,7 @@ toku_free(void *p) {
 
 void *
 toku_xmalloc(size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```

Comment:
just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)